docroot/modules/custom/mass_redirects/src/Plugin/QueueWorker/MassRedirectsRepoint.php

```diff
@@ -0,0 +1,161 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('database')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($data) {
+
+    $storage = $this->entityTypeManager->getStorage($data['usage_type']);
+
+    if (!$storage) {
+      return;
+    }
+
+    $entity = $storage->load($data['usage_id']);
+
+    if (!$entity) {
+      return;
+    }
+
+    if (!$entity->hasField($data['field'])) {
+      return;
+    }
+
+    $changed = FALSE;
+    $options = ['absolute' => TRUE];
+
+    $field_name = $data['field'];
+    $list = $entity->get($field_name);
+    $uri_old = 'entity:' . $data['from_type'] . '/' . $data['from_id'];
+    $uri_new = 'entity:' . $data['from_type'] . '/' . $data['to_id'];
+    foreach ($list as $delta => $item) {
+      dump(get_class($item));
+      switch (get_class($item)) {
+        case DynamicLinkItem::class:
+
+          $values[$delta] = $item->getValue();
+          $item_uri = $item->get('uri')->getString();
+          $item_uri_path = parse_url($item_uri, PHP_URL_PATH);
+          if ($item_uri == $uri_old) {
+            $values[$delta]['uri'] = $uri_new;
+            $changed = TRUE;
+          }
+          elseif ($item_uri_path == Url::fromUri($uri_old)->toString()) {
+            $values[$delta]['uri'] = Url::fromUri($uri_new, $options)->toString();
+            $changed = TRUE;
+          }
+          break;
+        case EntityReferenceItem::class:
+          $values[$delta] = $item->getValue();
+          if ($item->get('target_id')->getString() == $data['from_id']) {
+            $values[$delta]['target_id'] = $data['to_id'];
+            $changed = TRUE;
+          }
+          break;
+        case TextLongItem::class:
+        case TextWithSummaryItem::class:
+          $values[$delta] = $item->getValue();
+          // First check for the entity ID
+          if (str_contains($item->getString(), $data['from_id'])) {
+            $replaced = str_replace($data['from_id'], $data['to_id'], $item->getString());
+            $values[$delta]['value'] = $replaced;
+            $changed = TRUE;
+          }
+          // Next check for the link. We want relative links not
+          // absolute so domain mismatch isn't an issue.
+          if (str_contains($item->getString(), Url::fromUri($uri_old)->toString())) {
+            $replaced = str_replace(Url::fromUri($uri_old)->toString(), Url::fromUri($uri_new)->toString(), $item->getString());
+            $values[$delta]['value'] = $replaced;
+            $changed = TRUE;
+          }
+          break;
+      }
+
+      // Update the field values if any changes were made.
+      if ($changed) {
+        $entity->set($field_name, $values);
+        $entity->save();
```

Comment:
Its not a huge deal, but it would be faster to edit all the nodes and then save all the nodes. Its faster because some nodes have multiple field changes and those would get saved once.

Reply:
I'm passing one set of data to each queue item. I don't think there will be a huge impact on the performance. Feel free to suggest changes.